VPN (Private Egress) documentation/diagram

LinuxBozo opened the pull request:
Still missing technical details of what will happen to establish connections:
 - [x] ikev1 only
 - [x] will provide pre-shared key (PSK) from our side
 - [x] will provide exported configuration for various devices (if available, raw params if not)
 - [x] Info that connection has to be established and maintained (monitored/healthcheck) from their side

LinuxBozo committed:
Flesh out initial version of VPN (Private Egress) document/diagram

LinuxBozo committed:
Further information updates for more technical details

jmcarp commented on content/docs/apps/experimental/private-egress.md:
```diff
@@ -0,0 +1,76 @@
+title: Communicating with private external services
+---
+
+You can set up all, or just a subset, of your applications to be able communicate privately and securely with external services in your agency infrastructure or with other cloud providers.
```
"to be able to communicate"

jmcarp commented on content/docs/apps/experimental/private-egress.md:
```diff
@@ -0,0 +1,76 @@
+
+**NOTE**: At this time, only IKEv1 key exchance is supported, so make sure that your endpoint supports this.
+
+To use this service, you must have an internet-routable IP address to use as the endpoint for the connection. If you have a firewall in place between the Internet and your endpoint, then you will have to open both ingress and egress on UDP 500, and Protocol 50, to make use of this service. If you are also using NAT behind your firewall, you will also have to enable UDP 4500.
```
Does AWS support DNS as well as static IPs for VPN?

LinuxBozo replied:
Just static IP address.

jmcarp commented on content/docs/apps/experimental/private-egress.md:
```diff
@@ -0,0 +1,76 @@
+cloud.gov has the ability to work with you to establish an isolated set of servers, or cells, where only your Organization can place applications that need access to services that you control on other networks. Your applications will still be accessible from the Internet and can still connect to any of our offered internal services, just like any other applications on the platform, but they will  ...
+
+{{< diagrams id-prefix="colocated-diagram" >}}
+graph LR;
```
I would rather not have to scroll to see the whole graph. WDYT about switching to `graph TB`?

LinuxBozo replied:
Switched, though it's not much better. Still have to scroll. Also, added ASG information.

LinuxBozo committed:
Update graph, fix typo

LinuxBozo committed:
Clarifying details on what we provide

jmcarp reviewed (dismissed):
LGTM, but I'll hold off on merging until @mogul approves also.

mogul committed:
Tons of content edits and minor diagram updates

Reworked text for clarity/brevity/directness, then adjust the names of entites in the diagram to match.

mogul committed:
Cross-link static and private egress docs

mogul reviewed (commented):
Made a bunch of edits pairing with Britta. Josh will review and merge.
